modules/assim.batch/R/pda.utils.R

```diff
@@ -655,16 +655,19 @@ pda.postprocess <- function(settings, con, params, pname, prior, prior.ind, burn
 
   ## create a new Posteriors DB entry
   now <- format(Sys.time(), "%Y-%m-%d %H:%M:%S")
-  pft.id <- db.query(paste0(
-    "SELECT id from pfts where name = '", settings$pfts$pft$name,"'"), con)
+
+  pft.id <- db.query(paste0("SELECT pfts.id FROM pfts, modeltypes WHERE pfts.name='", settings$pfts$pft$name, "' and pfts.modeltype_id=modeltypes.id and modeltypes.name='", settings$model$type, "'"), con)[['id']]
```

Comment:
I think you want to do a JOIN between these two tables, not use the WHERE pfts.modeltype_id=modeltypes.id